Dockerfile

```diff
@@ -3,48 +3,66 @@ ARG UBUNTU_VERSION=20.04
+  yarn global add node-gyp@9.0.0
 WORKDIR /app
 COPY build build
-COPY packages packages
+COPY packages/cardano-services/package.json packages/cardano-services/package.json
+COPY packages/cardano-services-client/package.json packages/cardano-services-client/package.json
+COPY packages/core/package.json packages/core/package.json
+COPY packages/crypto/package.json packages/crypto/package.json
+COPY packages/dapp-connector/package.json packages/dapp-connector/package.json
+COPY packages/e2e/package.json packages/e2e/package.json
+COPY packages/golden-test-generator/package.json packages/golden-test-generator/package.json
+COPY packages/governance/package.json packages/governance/package.json
+COPY packages/hardware-ledger/package.json packages/hardware-ledger/package.json
+COPY packages/hardware-trezor/package.json packages/hardware-trezor/package.json
+COPY packages/input-selection/package.json packages/input-selection/package.json
+COPY packages/key-management/package.json packages/key-management/package.json
+COPY packages/ogmios/package.json packages/ogmios/package.json
+COPY packages/projection/package.json packages/projection/package.json
+COPY packages/projection-typeorm/package.json packages/projection-typeorm/package.json
+COPY packages/tx-construction/package.json packages/tx-construction/package.json
+COPY packages/util/package.json packages/util/package.json
+COPY packages/util-dev/package.json packages/util-dev/package.json
+COPY packages/util-rxjs/package.json packages/util-rxjs/package.json
+COPY packages/wallet/package.json packages/wallet/package.json
+COPY packages/web-extension/package.json packages/web-extension/package.json
 COPY scripts scripts
 COPY .yarn .yarn
-COPY \
-  .eslintrc.js \
-  .prettierrc \
-  .yarnrc.yml \
-  complete.eslintrc.js \
-  eslint.tsconfig.json \
-  package.json \
-  tsconfig.json \
-  yarn.lock \
-  yarn-project.nix \
-  /app/
+COPY .eslintrc.js .prettierrc .yarnrc.yml complete.eslintrc.js eslint.tsconfig.json package.json tsconfig.json yarn.lock yarn-project.nix ./
 
 FROM nodejs-builder as cardano-services-builder
 RUN yarn --immutable --inline-builds
-RUN NODE_OPTIONS=--max_old_space_size=10240 yarn build
+COPY packages packages
+RUN \
+  echo "export const unused = 'unused';" > packages/e2e/src/index.ts &&\
+  NODE_OPTIONS=--max_old_space_size=10240 yarn build
 
 FROM nodejs-builder as cardano-services-production-deps
 RUN yarn workspaces focus --all --production
 
 FROM ubuntu-nodejs as cardano-services
 COPY --from=cardano-services-production-deps /app/node_modules /app/node_modules
 COPY --from=cardano-services-production-deps /app/packages/cardano-services/node_modules /app/packages/cardano-services/node_modules
-COPY --from=cardano-services-production-deps /app/packages/projection-typeorm/node_modules /app/packages/projection-typeorm/node_modules
+COPY --from=cardano-services-production-deps /app/packages/core/node_modules /app/packages/core/node_modules
```

Comment:
Maybe we should copy node_modules from all packages?

Reply:
Not all packages have production dependencies.

To double check which one of them requires its own `node_modules` directory I've added following command (temporary, only to investigate this).

```
#39 [cardano-services-production-deps 2/2] RUN find . -name node_modules | grep -ve "^\.\/node"
#39 sha256:741d6c81dc0e63565f48b91e07b16a7ac09ef678a787a0f0360f3222af7216a6
#39 0.283 ./packages/cardano-services/node_modules
#39 0.283 ./packages/core/node_modules
#39 0.283 ./packages/e2e/node_modules
#39 0.283 ./packages/web-extension/node_modules
#39 0.283 ./packages/input-selection/node_modules
#39 0.283 ./packages/wallet/node_modules
#39 0.283 ./packages/golden-test-generator/node_modules
#39 DONE 0.3s
```

Here we need to copy only those directories actually required by the backend.

The only one I'm not sure is `input-selection`: could you please confirm it is used only client side?

Reply:
Yes input-selection is only used client-side. The reason I suggested copying `node_modules` of all packages (or all probably packages that are used in the backend) is brittle - a change in dependencies can result in a broken build.